WangscapeGui/MainWindow.cpp

```diff
@@ -0,0 +1,203 @@
+    // cause it is is used to generate that filepath and it makes
+    // modifying output dir for the whole output a lot easier
+    const std::string image_dir = mOptions->relativeOutputDirectory;
+    boost::filesystem::create_directories(image_dir);
+    boost::filesystem::path p(image_dir);
+    for (const auto& preview_image : mPreviewImages)
+    {
+        p.append(preview_image.first);
+        if (!preview_image.second.second.save(QString(p.string().c_str())))
+        {
+            QMessageBox critical_error_message;
+            critical_error_message.critical(this, "Error during saving tilesets", QString("Couldn't save tileset image"));
+            return;
+        }
+        p.remove_filename();
+    }
+
+    mTilesetGenerator->metaOutput.writeAll(*mOptions);
+
+    QMessageBox success_message;
+    success_message.information(this, "Saved!", QString("Output has been saved successfully"));
+}
+
+void MainWindow::resetOptions()
+{
+    mOptions = std::make_shared<Options>(mOriginalOptions);
+    mOptionsEditor->setOptions(mOptions);
+}
+
+void MainWindow::loadOptionsFromFile()
+{
+    QString chosen_path = QFileDialog::getOpenFileName(this,
+                                                     tr("Select options file"));
+    if (chosen_path.isEmpty())
+        return;
+
+    mUi->currentOptionsPath->setText(QString(chosen_path));
+
+    // TODO(hryniuk): make OptionsManager throw custom exception, catch it here
+    // and show QMessageBox with a proper error message
+    auto& options_manager = getOptionsManager();
+    options_manager.loadOptions(chosen_path.toLocal8Bit().constData());
+    mOriginalOptions = options_manager.getOptions();
+
+    resetOptions();
+
+    resetTilesetGenerator();
+
+    mUi->comboBox->clear();
+    mPreviewImages.clear();
+    mScene->clear();
+}
+
+void MainWindow::displayTilesetPreview(const QString& name)
+{
+    const QImage preview_image = name.isEmpty() ? (mPreviewImages.begin())->second.second : mPreviewImages.at(name.toLocal8Bit().constData()).second;
+    const QPixmap pixmap = QPixmap::fromImage(preview_image);
+
+    mScene->clear();
+    mScene->addPixmap(pixmap);
+    mScene->setSceneRect(pixmap.rect());
+    mUi->tilesetPreview->setScene(mScene);
+}
+
+void MainWindow::openOptionsEditor()
+{
+    mOptionsEditor->show();
+    addDockWidget(Qt::RightDockWidgetArea, mOptionsEditor);
+}
+
+void MainWindow::initializePreviewArea()
+{
+    mUi->comboBox->addItems([this]()
+    {
+        QStringList list;
+        for (const auto name : mPreviewImages)
+        {
+            list << QString(name.first.c_str());
+        }
+        return list;
+    }());
+}
+
+QImage MainWindow::convertSfImageToQImage(const sf::Image& source_image)
+{
+
+    const sf::Vector2u output_image_size = source_image.getSize();
+    QImage output_image(output_image_size.x, output_image_size.y, QImage::Format_RGB32);
+
+    for (int y = 0; y < output_image_size.y; ++y)
```

Comment:
`unsigned int y` to match `output_image_size.y`; same for `x`.

Reply:
I think we should build it with `-Wall` enabled and probably, when all warnings will be fixed, with `-Werror`.

Reply:
We used to have a warning from `googletest` during `WangscapeTest` compilation, but in the latest `master` build that error seems to be gone :)